storage/reader.go

```diff
@@ -80,6 +83,110 @@ func (o *ObjectHandle) NewReader(ctx context.Context) (*Reader, error) {
 	return o.NewRangeReader(ctx, 0, -1)
 }
 
+// newRangeReaderWithGRPC creates a new Reader with the given range that uses
+// gRPC to read Object content.
+//
+// This is an experimental API and not intended for public use.
+func (o *ObjectHandle) newRangeReaderWithGRPC(ctx context.Context, offset, length int64) (r *Reader, err error) {
+	ctx = trace.StartSpan(ctx, "cloud.google.com/go/storage.Object.newRangeReaderWithGRPC")
+	defer func() { trace.EndSpan(ctx, err) }()
+
+	if o.c.gc == nil {
+		err = fmt.Errorf("handle doesn't have a gRPC client initialized")
+		return
+	}
+	if err = o.validate(); err != nil {
+		return
+	}
+
+	// A negative length means "read to the end of the object", but the
+	// read_limit field it corresponds to uses zero to mean the same thing. Thus
+	// we coerce the length to 0 to read to the end of the object.
+	if length < 0 {
+		length = 0
+	}
+
+	req := &storagepb.ReadObjectRequest{
```

Comment:
Does the golang library support pre-conditions (`ifMetagenerationMatch`, `ifGenerationMatch`) or reading an specific generation of an object?  Do you want to support those for gRPC too?

Reply:
@tritone ?

Reply:
Worked offline, this is now implemented!